### ISSUE TITLE
Installing v3.0 on Ubuntu 20.04.2 fails when Docker is not running

### ISSUE THREAD

**Issue by AlexTalker** [nestybox/sysbox#269]
```
alx% sudo apt install ./sysbox-ce_0.3.0-0.ubuntu-focal_amd64.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'sysbox-ce' instead of './sysbox-ce_0.3.0-0.ubuntu-focal_amd64.deb'
The following packages were automatically installed and are no longer required:
  linux-headers-5.4.0-60 linux-headers-5.4.0-60-generic linux-headers-5.4.0-65 linux-headers-5.4.0-65-generic linux-image-4.15.0-130-generic linux-image-5.4.0-60-generic
  linux-image-5.4.0-65-generic linux-modules-4.15.0-130-generic linux-modules-5.4.0-60-generic linux-modules-5.4.0-65-generic linux-modules-extra-4.15.0-130-generic
  linux-modules-extra-5.4.0-60-generic linux-modules-extra-5.4.0-65-generic linux-tools-5.4.0-60 linux-tools-5.4.0-60-generic linux-tools-5.4.0-65 linux-tools-5.4.0-65-generic wmdocker
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  sysbox-ce
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/26.7 MB of archives.
After this operation, 72.7 MB of additional disk space will be used.
Get:1 /tmp/sysbox-ce_0.3.0-0.ubuntu-focal_amd64.deb sysbox-ce amd64 0.3.0-0.ubuntu-focal [26.7 MB]
Preconfiguring packages ...
Selecting previously unselected package sysbox-ce.
(Reading database ... 497433 files and directories currently installed.)
Preparing to unpack .../sysbox-ce_0.3.0-0.ubuntu-focal_amd64.deb ...
Unpacking sysbox-ce (0.3.0-0.ubuntu-focal) ...
Setting up sysbox-ce (0.3.0-0.ubuntu-focal) ...
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
dpkg: error processing package sysbox-ce (--configure):
 installed sysbox-ce package post-installation script subprocess returned error exit status 2
Errors were encountered while processing:
 sysbox-ce
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

I expected the process to be a lot smoother.

**Comment by AlexTalker** [comment]
It seems that disabling Docker services prior to installation gives out this error, and enabling them back allows for successful completion. I think some explicit checks are needed for this case then.

**Comment by rodnymolina** [comment]
@AlexTalker, thanks for reporting this issue.

As documented in our [installation](https://github.com/nestybox/sysbox/blob/master/docs/user-guide/install.md#host-requirements) guide, Docker is currently a requisite for the Sysbox installer. And the reason for this is that Sysbox installer configures Docker to be able to interact with the Sysbox runtime.

Now, you do have a point in that the process should be smoother even if Docker is not installed; our installer should deal with this scenario in a more graceful manner. I will take a look at how to improve this.

**Comment by ctalledo** [comment]
> Now, you do have a point in that the process should be smoother even if Docker is not installed; our installer should deal with this scenario in a more graceful manner. I will take a look at how to improve this.

This will be fixed in the upcoming release. 

The problem is that the installer is assuming that Docker is running rather than first checking whether it is or not.

Thanks @AlexTalker for reporting the issue.

**Comment by AlexTalker** [comment]
@rodnymolina The point is not that Docker is not installed(it is) but the fact that installation states "do docker bla-bla-bla" but does not state that you shall keep docker service up for time of installation(especially in the context of following "without Docker restart" section...

**Comment by AlexTalker** [comment]
The reason I tried the update in the first place is that I wanted to test proposed in #186 solution,
but it seems that things just go harder from there.

I see that your Ubuntu Focal image with systemd boots systemd successfully,
but mine based on latest Manjaro does not and I see the following:
```
[!!!!!!] Failed to allocate manager object.
```

I see that the difference between the Manjaro(v247) and the Ubuntu(v245) is 2 releases AND building options:
Ubuntu:
```
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
``` 
Manjaro:
```
+PAM +AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```

Could you please give me a hint if there's some incompatibility issue or this needs to be a separate issue?

@rodnymolina @ctalledo 

I ought to clarify that systemd process keeps on living on Manjaro, it just has no use at all and `systemctl` and such deny that PID 1 is indeed it.

I use image https://hub.docker.com/r/manjarolinux/base and updating all of the packages.

**Comment by AlexTalker** [comment]
Regarding the issue in previous my comment,
I dig into it a little bit more and found that you have https://github.com/nestybox/dockerfiles/tree/master/archlinux-systemd image.

So I pulled it, ran and it works too. I scratched my head a bit and found out that difference is down to minors - this image too has 247 Systemd but it's 247.2 and Manjaro is 247.6. So I ran `pacman -Syu` and it gave me 248, after which container does no longer starts:
```
Welcome to Arch Linux!


Failed to create /init.scope control group: Operation not permitted

Failed to allocate manager object: Operation not permitted

[!!!!!!] Failed to allocate manager object.

Exiting PID 1...
``` 

So I see my image and Arch have an issue in common...

Guess for now I'm gonna try to build my image without update but that might gave out a broken environment for me which is disappointing.

Shall I fill a separate issue on it then?

Now that it 100% reproducible on your original images?

**Comment by ctalledo** [comment]
Hi @AlexTalker 

> Shall I fill a separate issue on it then?

Yes that make sense since it's a different problem that the one being tracked by this current issue.

I've gone ahead an created issue #273 to track it. See there for an initial analysis of the problem.

**Comment by ctalledo** [comment]
> I've gone ahead an created issue #273 to track it. See there for an initial analysis of the problem.

My apologies, I had missed the fact that you created #271 already. I've closed #271 so we can track it in #273 because it has more info on the root cause.

**Comment by ctalledo** [comment]
Closing this issue as the Sysbox installer no longer requires Docker be present on the host.
